blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/AbstractPipelineImpl.java

```diff
@@ -143,7 +142,11 @@ public Link getLink() {
 
     @Override
     public String getFullName(){
-        return job.getFullName();
+        // Lets make sure that the fullName is properly encoded etc.
+        // The name part of the fullName can contain unencoded characters
+        // if the Job was programmatically created, causing screwups
+        // when mapping those URLs to Blue Ocean web/rest URLs. See JENKINS-41425.
+        return BlueUrlTokenizer.reconstructJobNamePath(job.getName(), job.getFullName());
```

Comment:
I think we need to override implementation of job.getFullName() and in there do safe-url-encoding, that is ensure we are not double encoding. For encoding perhaps we should use Util.rawEncode().